components/CreateQuizForm.tsx

```diff
@@ -70,99 +89,172 @@ export default function CreateQuizForm({ session }: CreateQuizFormProps) {
-                    overviewRulerLanes: 0,
-                    renderLineHighlight: "none",
-                  }}
-                />
-              </div>
-            </FormInput>
-
-            <FormInput
-              key="output"
+            </div>
+          </FormInput>
+
+          <FormInput
+            key="output"
+            name="output"
+            label="Target output"
+            sublabel="What should the user make the code output?"
+          >
+            <input
+              id="output"
               name="output"
-              label="Target output"
-              sublabel="What should the user make the code output?"
-            >
-              <input
-                id="output"
-                name="output"
-                type="text"
-                className="bg-gray-600 bg-opacity-25 text-white focus:ring-4 focus:ring-indigo-500 focus:ring-opacity-50 block w-full sm:text-sm border-0 rounded-xl py-4"
-                value={output}
-                onChange={(e) => setOutput(e.target.value)}
-                required={true}
-              />
-            </FormInput>
-          </div>
+              type="text"
+              className="block w-full py-4 text-white bg-gray-600 bg-opacity-25 border-0 focus:ring-4 focus:ring-indigo-500 focus:ring-opacity-50 sm:text-sm rounded-xl"
+              value={output}
+              onChange={(e) => setOutput(e.target.value)}
+              required={true}
+            />
+          </FormInput>
         </div>
+      </form>
+    ),
+    [FormStep.Solution]: (
+      <div>
+        <div className="flex">
+          <p className="text-white">
+            Complete your quiz and we'll include the solution!
+          </p>
+        </div>
+        <PreviewQuiz
+          key={startCode}
+          quiz={{
+            language: "typescript",
+            description,
+            solution,
+            start_code: startCode,
+            target_output: output,
+          }}
+          session={session}
+          profile={profile}
+          onSolution={(solution) => {
+            setSolution(solution);
+          }}
+        />
       </div>
+    ),
+    [FormStep.Explanation]: (
+      <div>
+        <div className="flex">
+          <p className="text-white">
+            If you'd like, you can create a guided explanation of the solution
+            code.
+          </p>
+        </div>
+        <ExplanationForm solution={solution} onChange={setExplanation} />
+      </div>
+    ),
+  };
 
+  return (
+    <>
+      <div className={classNames(formStep !== FormStep.Task && "hidden")}>
+        {steps[FormStep.Task]}
+      </div>
+      <div className={classNames(formStep !== FormStep.Solution && "hidden")}>
+        {steps[FormStep.Solution]}
+      </div>
+      <div
+        className={classNames(formStep !== FormStep.Explanation && "hidden")}
+      >
+        {steps[FormStep.Explanation]}
+      </div>
+      {/* {steps[formStep]} */}
```

Comment:
I was originally just conditionally rendering these steps like on line 213 here, but then the state for a step would get blown away if you switched to a different step and came back, so i changed it to just set display none on inactive steps to preserve their state